infra/src/main/java/com/redhat/service/bridge/infra/models/dto/ProcessorDTO.java

```diff
@@ -3,9 +3,10 @@
 import java.util.Objects;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.redhat.service.bridge.infra.logging.CustomerIdProvider;
 import com.redhat.service.bridge.infra.models.processors.ProcessorDefinition;
 
-public class ProcessorDTO {
+public class ProcessorDTO implements CustomerIdProvider {
```

Comment:
Is this really needed?

Reply:
Well..... it's used by the `shard-operator` logging where the `customerId` is _dynamic_ (i.e. a property of the resource being provisioned/deprovisioned etc) and cannot be _statically_ set as with the `executor` or `ingress`.

@robpblake wanted to keep logging of `customerId` transparent to the developer; so in the `shard-operator` we [intercept](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/541/files#diff-47ec8fe50ec7f971a1f568740548b00977a1bcd23747d5287023e88b03070c69R27) the logging call and if the parameter implements `CustomerIdProvider` we extract it and add to the log. Originally I [passed](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/541#discussion_r836792510) it as an additional logging parameter but it made the developer have to do more work.